src/platform/Darwin/PlatformManagerImpl.cpp

```diff
@@ -53,6 +53,27 @@ CHIP_ERROR PlatformManagerImpl::_InitChipStack()
     return err;
 }
 
+CHIP_ERROR PlatformManagerImpl::_StartEventLoopTask()
+{
+    dispatch_resume(mWorkQueue);
+
+    return CHIP_NO_ERROR;
+};
+
+CHIP_ERROR PlatformManagerImpl::_StopEventLoopTask()
+{
+    dispatch_sync(mWorkQueue, ^{
+        dispatch_suspend(mWorkQueue);
+    });
```

Comment:
Once RunEventLoop() below is fixed to be blocking, `StopEventLoopTask` has to be able to signal to that to permit it to exit the event loop.

Reply:
This is something I would like to figure out in a followup. The current code that uses `RunEventLoop` is kind of doing strange things on darwin since it blocks because of the shell engine and I would like to spend more time when I can fixing that.